js/entry.js

```diff
@@ -53,6 +53,10 @@ ipc.on(messages.REQUEST_WINDOW_STATE, () => {
   ipc.send(messages.RESPONSE_WINDOW_STATE, windowStore.getState().toJS())
 })
 
+ipc.on(messages.REQUEST_MENU_DATA_FOR_WINDOW, () => {
+  ipc.send(messages.RESPONSE_MENU_DATA_FOR_WINDOW, windowStore.getState().toJS())
```

Comment:
I'd really like to avoid this kind of state transfer in general and I don't think it's going to be correctly synced anyway because this updates on appState changes, not windowState changes. This will be much easier once we move webContents management to the main process and if the only purpose of this is to get close frames I would just leave it out for now. If we really need it now then I think appState should track WINDOW_CLOSE_FRAME, WINDOW_UNDO_CLOSED_FRAME. That will mean some state duplication between windowState and appState, but still better than relaying windowState imo. There is also likely a performance issue doing this on every app state change.